activation/handler.go

```diff
@@ -220,27 +247,40 @@ func (h *Handler) SyntacticallyValidateAtx(ctx context.Context, atx *types.Activ
 			return nil, fmt.Errorf("atx layer (%v) must be after positioning atx layer (%v)", atx.PubLayerID, posAtx.PubLayerID)
 		}
 		if d := atx.PubLayerID.Difference(posAtx.PubLayerID); d > h.layersPerEpoch {
-			return nil, fmt.Errorf("expected distance of one epoch (%v layers) from pos atx but found %v", h.layersPerEpoch, d)
+			return nil, fmt.Errorf("expected distance of one epoch (%v layers) from positioning atx but found %v", h.layersPerEpoch, d)
 		}
 		baseTickHeight = posAtx.TickHeight()
 	} else {
 		publicationEpoch := atx.PublishEpoch()
 		if !publicationEpoch.NeedsGoldenPositioningATX() {
-			return nil, fmt.Errorf("golden atx used for atx in epoch %d, but is only valid in epoch 1", publicationEpoch)
+			return nil, fmt.Errorf("golden atx used for positioning atx in epoch %d, but is only valid in epoch 1", publicationEpoch)
 		}
 	}
 
 	expectedChallengeHash, err := atx.NIPostChallenge.Hash()
 	if err != nil {
-		return nil, fmt.Errorf("failed to compute NIPost's expected challenge hash: %v", err)
+		return nil, fmt.Errorf("failed to compute NIPost's expected challenge hash: %w", err)
 	}
 
 	h.log.WithContext(ctx).With().Info("validating nipost", log.String("expected_challenge_hash", expectedChallengeHash.String()), atx.ID())
 
-	pubKey := signing.NewPublicKey(atx.NodeID().ToBytes())
-	leaves, err := h.nipostValidator.Validate(*pubKey, atx.NIPost, *expectedChallengeHash, uint(atx.NumUnits))
+	commitmentATX := atx.CommitmentATX
+	if atx.CommitmentATX == *types.EmptyATXID {
```

Comment:
the syntactic check of CommitmentATX is scattered in 2, 3 different places. is it possible to re-organize SyntacticallyValidateAtx a little such that the expectation for CommitmentATX is in its own place/method?

Reply:
I will take a look at `SyntacticallyValidateAtx` and try to reduce its cyclomatic complexity by splitting it up in multiple methods, probably one for each field that is validated.

Reply:
I broke the method up into 3 individual methods, one for validating initial ATX, one for validating non-initial ATX and one that applies validations that are relevant for both.